src/services/codefixes/importFixes.ts

```diff
@@ -864,9 +866,14 @@ function codeFixActionToCodeAction({ description, changes, commands }: CodeFixAc
 
 function getAllExportInfoForSymbol(importingFile: SourceFile | FutureSourceFile, symbol: Symbol, symbolName: string, moduleSymbol: Symbol, preferCapitalized: boolean, program: Program, host: LanguageServiceHost, preferences: UserPreferences, cancellationToken: CancellationToken | undefined): readonly SymbolExportInfo[] | undefined {
     const getChecker = createGetChecker(program, host);
+    const isFileExcluded = preferences.autoImportFileExcludePatterns && getIsFileExcluded(host, preferences);
+    const moduleSymbolExcluded = isFileExcluded && moduleSymbol.declarations && isSourceFile(moduleSymbol.declarations[0]) && isFileExcluded(moduleSymbol.declarations[0]);
     return getExportInfoMap(importingFile, host, program, preferences, cancellationToken)
         .search(importingFile.path, preferCapitalized, name => name === symbolName, info => {
-            if (skipAlias(info[0].symbol, getChecker(info[0].isFromPackageJson)) === symbol && info.some(i => i.moduleSymbol === moduleSymbol || i.symbol.parent === moduleSymbol)) {
+            if (
+                skipAlias(info[0].symbol, getChecker(info[0].isFromPackageJson)) === symbol
+                && info.some(i => moduleSymbolExcluded || i.moduleSymbol === moduleSymbol || i.symbol.parent === moduleSymbol)

```

Comment:
Plot twist: all existing tests pass if you delete this entire `info.some(...)` part of the condition. That might be all that’s needed to fix the bug?

Reply:
I tried that and wasn't sure if that was the best fix, since I tracked it back to [here](https://github.com/microsoft/TypeScript/pull/52686/files#r1103313654), where it looked like it was intentionally added

Reply:
Actually, I remember something did break when I tried that-- `tests/cases/fourslash/server/autoImportReExportFromAmbientModule.ts` breaks with the `info.some` call removed. Since exporting ambient modules doesn't use filenames, I figured it would be alright to skip the check for ignored files

Reply:
```suggestion
                && (moduleSymbolExcluded || info.some(i => i.moduleSymbol === moduleSymbol || i.symbol.parent === moduleSymbol))
```